src/visitor.rs

```diff
@@ -326,6 +342,37 @@ pub trait Visitor<'a> {
         Ok(())
     }
 
+    fn visit_upsert(&mut self, update: Update<'a>) -> Result {
+        self.write("UPDATE ")?;
+
+        self.write("SET ")?;
+        self.visit_update_set(update.clone())?;
+
+        if let Some(conditions) = update.conditions {
+            self.write(" WHERE ")?;
+            self.visit_conditions(conditions)?;
+        }
+
+        Ok(())
+    }
+
+    fn visit_update_set(&mut self, update: Update<'a>) -> Result {
+        let pairs = update.columns.into_iter().zip(update.values.into_iter());
+        let len = pairs.len();
+
+        for (i, (key, value)) in pairs.enumerate() {
+            self.visit_column(key)?;
+            self.write(" = ")?;
+            self.visit_expression(value)?;
+
+            if i < (len - 1) {
+                self.write(", ")?;
+            }
+        }
+
+        Ok(())
+    }
```

Comment:
Is there any difference between this and [`visit_update`](https://github.com/prisma/quaint/blob/34d170cc04c50cc8c4f3ac01d638720f404fb49b/src/visitor.rs#L295-L327) ?

Reply:
Ah, we don't visit the table in this upsert version, since we already visited it for the insert

Reply:
Right. Its kind of annoying but I couldn't see another design